classes/services/PKPPublicationService.inc.php

```diff
@@ -624,22 +652,39 @@ public function publish($publication)
 
         // Log an event when publication is published. Adjust the message depending
         // on whether this is the first publication or a subsequent version
-        if (count($submission->getData('publications')) > 1) {
-            $msg = $newPublication->getData('status') === PKPSubmission::STATUS_SCHEDULED ? 'publication.event.versionScheduled' : 'publication.event.versionPublished';
-        } else {
-            $msg = $newPublication->getData('status') === PKPSubmission::STATUS_SCHEDULED ? 'publication.event.scheduled' : 'publication.event.published';
-        }
-        SubmissionLog::logEvent(Application::get()->getRequest(), $submission, SubmissionEventLogEntry::SUBMISSION_LOG_METADATA_PUBLISH, $msg);
+        $availableMessages = [
```

Comment:
It's tough to review code when there are new concepts mixed in with general clean-up; can you keep clean-up to separate PRs when working on something like this? I see the long lines and ternaries you're working to clean up, but the array lookup syntax in `$availableMessages` isn't clearer to me.